clouddriver-kubernetes/src/integration/java/com/netflix/spinnaker/clouddriver/kubernetes/it/DeployManifestIT.java

```diff
@@ -826,17 +824,24 @@ private void deployAndWaitStable(
             () -> {
               Response fcrWaitResp =
                   given().log().uri().get(baseUrl() + "/cache/kubernetes/manifest");
-              JsonPath jsonPath = fcrWaitResp.jsonPath();
+              fcrWaitResp.then().log().body(false);
               List<Object> list =
-                  jsonPath.get(
-                      "findAll { it -> it.details.account == \""
-                          + ACCOUNT1_NAME
-                          + "\" && it.details.location == \""
-                          + targetNs
-                          + "\" && it.details.name == \""
-                          + objectName
-                          + "\" && it.processedTime > -1 }");
-              return list != null && !list.isEmpty();
+                  Stream.of(fcrWaitResp.as(Map[].class))
```

Comment:
> List filter code for the above request was rewritten from rest-assured jsonpath to plain java code

Was this part of the fix or just rewritten to improve debuggability?

Reply:
This is the actual change that hopefully prevents intermittent failures. For some reason, when filtering the on demand cache results looking for the desired object using rest-assured json path, the object was not always returned by the filter even if it was present in the response. I assume it might be related to the `processedTime` comparison having to be treated as a `long`.

So I changed the filter to use `Object.equals` for the other fields and treating `processedTime` as a java `Number` to guarantee number comparisons in all cases. On my debugging, if `processedTime` was `-1` it was returned as `Integer` and if it had an actual time it was a `Long`.

Reply:
>  On my debugging, if processedTime was -1 it was returned as Integer and if it had an actual time it was a Long.

Oh, so strange! Thanks for the explanation.